CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ option(WAMR_INTERPRETER_MODE "Turns interpreter on / off (vs. AoT)" OFF)
 
 # SGX functionality
 # These options customise the SGX features _provided_ SGX is found
+option(FAASM_SGX_ENABLED "Enable SGX mode" OFF)
```

Comment:
I thought that `find_pacakage(SGX)` would just set `SGX_FOUND=false` if SGX wasn't installed, and that's how you do an SGX-free build. This autodetects the environment rather than having to change the option flag, which seems a bit nicer.